php-frontend/src/main/java/org/sonar/php/parser/TreeFactory.java

```diff
@@ -0,0 +1,1394 @@
+        result = ((MemberAccessTreeImpl) memberAccess).complete(result);
+
+      } else if (memberAccess.is(Kind.ARRAY_ACCESS)) {
+        result = ((ArrayAccessTreeImpl) memberAccess).complete(result);
+
+      } else if (memberAccess.is(Kind.FUNCTION_CALL)) {
+        result = ((FunctionCallTreeImpl) memberAccess).complete(result);
+      }
+    }
+
+    return result;
+  }
+
+  public VariableTree lexicalVariable(Optional<InternalSyntaxToken> ampersandToken, VariableIdentifierTree variableIdentifier) {
+    return ampersandToken.isPresent()
+      ? new ReferenceVariableTreeImpl(ampersandToken.get(), variableIdentifier)
+      : variableIdentifier;
+  }
+
+  public LexicalVariablesTree lexicalVariables(
+      InternalSyntaxToken useToken, InternalSyntaxToken openParenthesis,
+      VariableTree variable, Optional<List<Tuple<InternalSyntaxToken, VariableTree>>> variableRest,
+      InternalSyntaxToken closeParenthesis
+  ) {
+    return new LexicalVariablesTreeImpl(useToken, openParenthesis, separatedList(variable, variableRest), closeParenthesis);
+  }
+
+  public FunctionCallTree internalFunction1(InternalSyntaxToken issetToken, InternalSyntaxToken openParenthesis, ExpressionTree expression, Optional<List<Tuple<InternalSyntaxToken, ExpressionTree>>> expressionRest, InternalSyntaxToken closeParenthesis) {
+    return new FunctionCallTreeImpl(
+      new IdentifierTreeImpl(issetToken),
+      openParenthesis,
+      separatedList(expression, expressionRest),
+      closeParenthesis);
+  }
+
+  public FunctionCallTree internalFunction2(InternalSyntaxToken functionNameToken, InternalSyntaxToken openParenthesis, ExpressionTree expression, InternalSyntaxToken closeParenthesis) {
+    return new FunctionCallTreeImpl(
+      new IdentifierTreeImpl(functionNameToken),
+      openParenthesis,
+      new SeparatedList(ImmutableList.of(expression), ImmutableList.<InternalSyntaxToken>of()),
+      closeParenthesis);
+  }
+
+  public FunctionCallTree internalFunction3(InternalSyntaxToken includeOnceToken, ExpressionTree expression) {
+    return new FunctionCallTreeImpl(
+      new IdentifierTreeImpl(includeOnceToken),
+      new SeparatedList(ImmutableList.of(expression), ImmutableList.<InternalSyntaxToken>of()));
+  }
+
+  public ArrayPairTree arrayPair1(ExpressionTree expression, Optional<Tuple<InternalSyntaxToken, ExpressionTree>> pairExpression) {
+    if (pairExpression.isPresent()) {
+      return new ArrayPairTreeImpl(expression, pairExpression.get().first(), pairExpression.get().second());
+    }
+    return new ArrayPairTreeImpl(expression);
+  }
+
+  public ArrayPairTree arrayPair2(ReferenceVariableTree referenceVariableTree) {
+    return new ArrayPairTreeImpl(referenceVariableTree);
+  }
+
+  public SeparatedList<ArrayPairTree> arrayInitialiserList(ArrayPairTree firstElement, Optional<List<Tuple<InternalSyntaxToken, ArrayPairTree>>> restElements, Optional<InternalSyntaxToken> trailingComma) {
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Split this 205 characters long line (which is greater than 180 authorized). [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS00103)